### PR TITLE
Fix vehicle and graph tab of ride window and options window

### LINKED ISSUE #1003: Altitude markings are wrong on tracked ride graphs
scr2

### LINKED ISSUE #1005: Options window opened below extra tools windows
When using one of the extra tools, like the Scenario Editor or the RollerCoaster Designer that uses a big window (like the Object selection list in the Scenario Editor), and then trying the open the Options window will open the Options window, but underneath the big window (which is revealed when you click Next), rather than in the foreground.

### LINKED ISSUE #1016: Adding extra block sections to existing rides doesn't change max number of trains
To reproduce:
- Build a tracked ride with n block sections
- Check the vehicle tab, it will allow you to put n-1 trains on the track
- Edit the ride to add more block section
- Check the vehicle tab. The maximum number of trains is not updated.

Selecting a train type from the dropdown in the vehicle tab does update the maximum number of trains.

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Fixes #1016.
Looks like this little function was missed when reverseing this function.
Also fixed #1005.
Also fixed #1003 as well.

### CODE REVIEW

File: src/windows/options.c [hunk 1]
```diff
@@ -221,7 +221,7 @@ void window_options_open()
 	if (w != NULL)
 		return;
 
-	w = window_create_centred(WW, WH, (uint32*)window_options_events, WC_OPTIONS, 0);
+	w = window_create_centred(WW, WH, (uint32*)window_options_events, WC_OPTIONS, WF_STICK_TO_FRONT);
```
Comment (Gymnasiast) [inline src/windows/options.c]: This has the side effect that closing all windows via a shortcut (Shift + Backspace) doesn't close the Options window.

**Comment by duncanspumpkin** [comment]
Will a call to bring_to_front after window creation work. I'm loathe to change how any other windows operate.

**Comment by Gymnasiast** [comment]
No, that doesn't seem to work.

**Comment by duncanspumpkin** [comment]
I'll think of something tonight.

**Comment by duncanspumpkin** [comment]
So i had a look the object selection window isn't meant to be a stick to front window so i changed that. The scenario selection window is meant to be a stick to front but i've made it so that it no longer is. I think that causes the least amount of changes.

**Pull request merged**